### PR TITLE
Add UC OSS documentation for the model registry

### PR DESCRIPTION
**Pull request by kriscon-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/kriscon-db/mlflow/pull/13107?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13107/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13107
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Added UC OSS documentation.

### How is this PR tested?

- [x ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [x ] Yes. I've updated:
  - [x ] Examples
  - [ ] API references
  - [ x] Instructions

#### Is this a user-facing change?

- [x ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: docs/source/model-registry.rst [hunk 1]
```diff
@@ -230,6 +230,21 @@ token = <your Databricks shard access token>
     mlflow.set_registry_uri("databricks-uc://my-databricks-shard1")
 
 
+OSS Unity Catalog Model Registry
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To use an `OSS Unity Catalog <https://www.unitycatalog.io/>` server as your MLflow model registry, specify the UC server address in the MLflow registry URI in the format ``"uc:http://localhost:8080"``.  If your unity catalog server is configured to use authentication, set the environmental variable ``"MLFLOW_UC_OSS_TOKEN"``.
```
Comment (kriscon-db) [inline docs/source/model-registry.rst:236]: This doesn't appear to hyper link `OSS Unity Catalog`.  Any idea what the issue here is?

Comment (harupy) [reply docs/source/model-registry.rst:236]: /ai What's missing?

Comment (harupy) [reply docs/source/model-registry.rst:236]:
```suggestion
To use an `OSS Unity Catalog <https://www.unitycatalog.io/>`_ server as your MLflow model registry, specify the UC server address in the MLflow registry URI in the format ``"uc:http://localhost:8080"``.  If your unity catalog server is configured to use authentication, set the environmental variable ``"MLFLOW_UC_OSS_TOKEN"``.
```

`_` is missing.

File: docs/source/model-registry.rst [hunk 2]
```diff
@@ -230,6 +230,21 @@ token = <your Databricks shard access token>
...
+
+**Use the OSS unity catalog model registry using a bearer token**
+
+.. code-block:: python
+
+    import mlflow
+    import os
+
+    mlflow.set_registry_uri("uc:http://localhost:8080")
+    os.environ["MLFLOW_UC_OSS_TOKEN"] = "<your OSS UC access token>"
```
Comment (harupy) [inline docs/source/model-registry.rst:246]: Hardcoding a token is not a good practice. Can we use `getpass`?

Comment (kriscon-db) [reply docs/source/model-registry.rst:246]: We can but I'm worried about making the cutoff for Friday as I have very little time left to be adding new code/testing before then.  We can add that as an improvement for after release.  UC OSS is still v0.2 right now and is hardly what I would consider production ready as it is.

Comment (kriscon-db) [reply docs/source/model-registry.rst:246]: Do you want that note added here @BenWilson2 ?

Comment (BenWilson2) [reply docs/source/model-registry.rst:246]: We don't need that full note, just a comment in the code block to remind users to set the env var. We remind people in other places in the docs by doing an assert on the existence of the key via something like:
```python
assert "MLFLOW_UC_OSS_TOKEN" in os.environ.keys()
```

Comment (harupy) [reply docs/source/model-registry.rst:246]: @kriscon-db 

We just need to add this.

```python
from getpass import getpass

os.environ["MLFLOW_UC_OSS_TOKEN"] = getpass(...)
```

We can improve this as a follow-up.

File: docs/source/model-registry.rst [hunk 3]
```diff
@@ -230,6 +230,21 @@ token = <your Databricks shard access token>
     mlflow.set_registry_uri("databricks-uc://my-databricks-shard1")
 
 
+OSS Unity Catalog Model Registry
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (harupy) [inline docs/source/model-registry.rst:234]:
```suggestion
OSS Unity Catalog Model Registry
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**